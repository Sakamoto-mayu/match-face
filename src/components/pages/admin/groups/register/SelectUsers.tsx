import UserSelectModal from "@/components/pages/admin/tasks/register/UserSelectModal";
// import UserInput from "./UserInput";

const SelectUsers = async() =>{
return(
    <div>
    <div className="flex">
    {/* <label htmlFor="user" className="leading-9 mr-3">▶ユーザーを選択する</label> */}
    <UserSelectModal/>
    </div>
    {/* <div>
        <UserInput/>
    </div> */}
    </div>
)
}

export default SelectUsers