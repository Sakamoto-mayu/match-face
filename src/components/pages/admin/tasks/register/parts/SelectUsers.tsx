import Input from "@/components/ui/Input";
import WhiteButton from "@/components/ui/button/WhiteButton";

const SelectUsers = () => {
  return (
    <div>
      <div className="flex items-center">
        <p>ユーザーを選択する</p>
        <WhiteButton label="追加" className="text-xs" />
      </div>
      <div>
        <Input
          id="search"
          className="border-2 border-light-gray text-xs p-1 w-96"
          readOnly
        />
      </div>
    </div>
  );
};

export default SelectUsers;
