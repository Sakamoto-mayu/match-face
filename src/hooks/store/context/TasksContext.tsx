"use client";

import React from "react";

// 職種フィルター用context
const JobsFilterContext = React.createContext<
  [string[], React.Dispatch<React.SetStateAction<string[]>>] | undefined
>(undefined);

export const JobsFilterProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [filterlist, setFilterList] = React.useState<string[]>([]);
  return (
    <JobsFilterContext.Provider value={[filterlist, setFilterList]}>
      {children}
    </JobsFilterContext.Provider>
  );
};

export const useJobsFilter = () => {
  const context = React.useContext(JobsFilterContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a Provider");
  }
  return context;
};
// ここまで
