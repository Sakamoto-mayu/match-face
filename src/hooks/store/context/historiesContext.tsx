'use client'
import { createContext, useContext } from "react";
import { Departments, Skills } from "@/types/admin/histories/admin_histories";
import { skills,departments } from "@/const/admin_histories";

export const SkillsContext = createContext<Skills|undefined>(undefined)
export const DepartmentsContext = createContext<Departments|undefined>(undefined)

// constからのimport(valueの値)をfetchで取得したデータに変える
// const skills = fetch('')
// const departments = fetch('')

export function HistoriesProvider({children}:{children:React.ReactElement}) {
    return (
        <SkillsContext.Provider value = {skills}>
        <DepartmentsContext.Provider value = {departments}>
            {children}
        </DepartmentsContext.Provider>
        </SkillsContext.Provider>
    )
}
export const useSkills = () => {
    const context = useContext(SkillsContext)
    if(context === undefined){
        throw new Error('skillsが見つかりません')
    }
    return context
}
export const useDepartments = () => {
    const context = useContext(DepartmentsContext)
    if(context === undefined){
        throw new Error('departmentsが見つかりません')
    }
    return context
}
