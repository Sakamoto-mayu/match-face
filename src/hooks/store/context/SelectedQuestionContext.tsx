"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { selectedQuestionReducer } from "../reducer/selectedQuestionReducer";
import { SelectReducerAction } from "@/types/admin/tasks/register/types";

const SelectedQuestionContext = createContext<
  [string[], Dispatch<SelectReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], SelectReducerAction>>(
    selectedQuestionReducer,
    []
  );
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
