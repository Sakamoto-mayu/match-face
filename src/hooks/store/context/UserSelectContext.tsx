"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { userSelectReducer } from "../reducer/userSelectReducer";
import { UserSelectReducerAction } from "@/types/admin/tasks/register/types";

const UserSelectContext = createContext<
  [string[], Dispatch<UserSelectReducerAction>] | undefined
>(undefined);

export const CounterProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], UserSelectReducerAction>>(
    userSelectReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useCounter' must be used in Provider.");
  return context;
};
