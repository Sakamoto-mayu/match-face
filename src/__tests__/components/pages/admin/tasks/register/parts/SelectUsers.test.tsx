import SelectUsers from "@/components/pages/admin/tasks/register/parts/SelectUsers";
import { UserSelectProvider } from "@/hooks/store/context/UserSelectContext";
import { render } from "@testing-library/react";

describe("SelectUsers.tsx", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.clearAllMocks();
  });

  it("レンダリング時", () => {
    const {container} = render(<UserSelectProvider><SelectUsers /></UserSelectProvider>);
    expect(container).toMatchSnapshot();
  });
});
