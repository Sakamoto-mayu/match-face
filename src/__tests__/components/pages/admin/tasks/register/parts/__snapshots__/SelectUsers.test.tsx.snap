// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectUsers.tsx レンダリング時 1`] = `
<div>
  <div
    class="mb-5"
  >
    <div
      class="flex items-center mb-2"
    >
      <p
        class="text-base w-40"
      >
        ▶ユーザーを選択する
      </p>
      <div
        class="text-xs ml-10"
      >
        <button
          class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
        >
          追加
        </button>
      </div>
    </div>
    <div>
      <div>
        <input
          class="border-2 border-light-gray text-xs p-1 w-full"
          id="search"
          readonly=""
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
